### ISSUE TITLE
cref's to public members become ambiguous when an internal member is also in scope

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#4033]
This is related to #77, and was raised here: https://stackoverflow.com/questions/31541390/visual-studio-2015-c-sharp-6-cant-compile-xml-comments-in-pcl-project.

Basically, cref's to public types are now ambiguous if there is an internal type in scope, resulting in a build warning for projects upgrading to C# 6.

 The case in the stack overflow question can be repro'd by building against C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework.NETPortable\v4.0\Profile\Profile47\mscorlib.dll, and writing the following (there's an internal class called "Type" in System.Reflection which is now ambiguous with the one in System):

``` C#
using System;
using System.Reflection;

/// <summary>
///     <see cref="Type"/>
/// </summary>
public class Class1
{
}

```

Which warns with (note filed #4031 for lack of namespace qualification in the warning):

```
CS0419  Ambiguous reference in cref attribute: 'Type'. Assuming 'Type', but could have also matched other overloads including 'Type'
```

I believe there were deliberate architectural decisions that resulted in this, but just wanted to get a bug filed to track the change in behavior.

**Comment by jaredpar** [comment]
Looks like a duplicate of #3500

**Comment by davkean** [comment]
Doesn't look like a dupe, #3500 is talking about the fact that sometimes internals are available and sometimes they are not. This is talking about, favoring public types over internal types in a cref.

**Comment by jaredpar** [comment]
I believe it is the same underlying issue.  In the cases where they are available they allow for conflicts.  These bugs seem to identify two different cases for the same issue.

**Comment by davkean** [comment]
Crefs to internals, as far as I am aware, is by design. They were added to maintain compat with the native compiler (turns out customers do this a lot to both internals and privates). While fixing #3500 to stop looking at internals unless there's InternalsVisibleTo to will fix the above problem, I don't believe it's the correct fix.

**Comment by julealgon** [comment]
Getting this warning on my side as well. My question is: how does one cleanly fix this besides creating a weird alias?

I see that the warning stops if I do this:
```csharp
using ProblematicClass = Namespace.ProblematicClass;
```
even if `Namespace` is already in the list of included namespace statements in the file.

**Comment by jaredpar** [comment]
Going to let #71442 be the primary here as that has the breaking change proposal attached to it.
